docs/artifacts.md

```diff
@@ -0,0 +1,97 @@
+Artifact Versioning and Tagging (1.0.0)
+=======================================
+
+Artifact Name Composition
+-------------------------
+
+### Filenames
+
+`<--artifact_name-->@<--semantic_version-->.<--file_extension-->`
+
+Example:
+
+`electron-bootloader@1.0.1+lto.debug.jtag.bin`
+
+#### Filename Composition
+
+- Artifact Name
+  - (Platfrom Family Name (_none_) | Platform Name (`electron`))
+  - Module Name (`bootloader`)
+- Semantic Version
+  - Major Version (`1`)
+  - Minor Version (`0`)
+  - Patch Version (`1`)
+  - [_OPTIONAL_] Prerelease Tag (_none_)
+  - [_OPTIONAL_] Build Metadata 1,2,...,n (`lto`, `debug`, `jtag`)
+- File Extension (`bin`)
+
+**Rules:**
+
+- Artifacts targeting all platforms within a given platform family, shall include the platform family name and not a specific platform name (i.e. `gen3-tinker@1.1.0.bin`)
+- Boolean build parameters are added to metadata when `true` and absent when `false`
+- Enumerated build parameter "values" are added to metadata
+
+**Seperators:**
+
+- `-` (_hyphen_) - Seperates platform and modules names
+- `@` (_at symbol_) - Seperates the artifact name from the semantic version
+- `.` (_period_) - Precedes the file extension
+
+> _**NOTE:** Seperators belonging to the [semantic versioning specification](https://semver.org/) will be nested within the binary name. Those seperators are NOT listed here._
+
+### Folder Hierarchy
+
+`<--evaluated_semantic_version-->/<--platform-->/<--binary_descriptor-->`
+
+Example:
+
+`1.0.1/electron/debug`
+
+#### Hierarchical Composition
+
+- Evaluated Semantic Version*
+  - Major Version (`1`)
+  - Minor Version (`0`)
+  - Patch Version (`1`)
+  - [_OPTIONAL_] Pre-release Tag (_none_)
+- Platform Name (`electron`)
+- Binary Descriptor (`debug`)
+
+### Zip Artifacts
+
+`<--company-->_<--product-->@<--evaluated_semantic_version-->+<--platform-->.<--content-->.<--file_extension-->`
+
+Example:
+
+`particle_device-os@1.0.1+electron.binaries.zip`
```

Comment:
I have a few suggestions:
- Let's remove the company name from `.zip` artifact names to make them shorter.
- Let's use `-` (a dash) everywhere in artifact names as a separator.
- Do we really need the `binaries` tag? Can we omit it by default for archives that contain Device OS binaries?
- I'd also move the platform name to the artifact name, so that the names would look like `device-os-electron@1.0.1.zip`.

Reply:
I particularly like the company name there, so when people download it and leave it in their downloads folder for three months they will know what it is. Also, if you download multiple `.zip` files, they will all be in one place (alphabetically speaking).
I'm not in love with the name `binaries` either. However `electron` gives you the folders `release`, `debug` and `test`, while 'electron binaries` only gives you the contents of the `release` folder (i.e. the files you would actually put on the device).